researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java

```diff
@@ -579,6 +590,24 @@ public void uploadTaskResult(Context context, @NonNull TaskResult taskResult) {
             }
         }
 
+        if (lastLoadedTaskGuid == null) {
+            logger.error("lastLoadedTaskGuid must be set for this task to complete and " +
+                    "set finishedOn and startedOn dates, and clientData on the bridge server");
+        } else {
+            // TODO: make GUID available in ScheduledActivity constructor and get rid of this gson nonsense
```

Comment:
I don't have permission or really know how to change the swagger auto-generated code, but we need to be able to set a ScheduledActivity's GUID (it is a private member variable) somehow without doing this GSON reflection work-around.